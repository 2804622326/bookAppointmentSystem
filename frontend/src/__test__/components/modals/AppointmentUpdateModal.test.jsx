import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import AppointmentUpdateModal from '../../../components/modals/AppointmentUpdateModal.jsx';

jest.mock('react-datepicker', () => (props) => (
  <input data-testid='picker' onChange={(e) => props.onChange(new Date(e.target.value))} />
));

test('updates appointment when submitted', () => {
  const appointment = {
    appointmentDate: '2024-01-01',
    appointmentTime: '10:00',
    reason: 'Check',
  };
  const update = jest.fn();
  const { asFragment } = render(
    <AppointmentUpdateModal show handleClose={() => {}} appointment={appointment} handleUpdate={update} />
  );
  fireEvent.change(screen.getAllByTestId('picker')[0], { target: { value: '2024-01-02' } });
  fireEvent.change(screen.getAllByTestId('picker')[1], { target: { value: '11:00' } });
  fireEvent.change(screen.getByPlaceholderText(/enter reason/i), { target: { value: 'Updated' } });
  fireEvent.click(screen.getByText(/save update/i));
  expect(update).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
