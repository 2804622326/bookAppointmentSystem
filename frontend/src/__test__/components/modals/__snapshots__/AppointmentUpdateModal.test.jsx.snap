// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updates appointment when submitted 1`] = `<DocumentFragment />`;
